### PR TITLE
Support Presign options

### PR DESCRIPTION
**Pull request by sandeep-paliwal**
Allow presign options to be specified in TVM presign request

- Permissions can now be passed to TVM to override default of read only permission

## How Has This Been Tested?
unit tests and manual tests

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ x] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [x ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x ] I have read the **CONTRIBUTING** document.
- [x ] I have added tests to cover my changes.
- [x ] All new and existing tests passed.

### CODE REVIEW

File: lib/impl/AzurePresignTvm.js [hunk 1]
```diff
@@ -45,7 +46,7 @@ class AzurePresignTvm extends Tvm {
     const expiryTime = new Date()
     expiryTime.setSeconds(expiryTime.getSeconds() + params.expiryInSeconds)
 
-    const permissions = azure.BlobSASPermissions.parse('r')
+    const permissions = azure.BlobSASPermissions.parse(params.permissions)
```
Comment (moritzraho) [inline lib/impl/AzurePresignTvm.js:49]: if params.permissions is invalid => we would get a 500, maybe we can have the validation schema take care of this (with a regex or so)

File: lib/impl/AzurePresignTvm.js [hunk 2]
```diff
@@ -29,7 +29,10 @@ class AzurePresignTvm extends Tvm {
     this._addToValidationSchema('azureStorageAccount')
     this._addToValidationSchema('azureStorageAccessKey')
     this._addToValidationSchema('blobName')
-    this._addToValidationSchema('expiryInSeconds', joi.number().integer().min(30).max(18000).required())
+    this._addToValidationSchema('expiryInSeconds', joi.number().integer().min(2).max(86400).required())
```
Comment (purplecabbage) [inline lib/impl/AzurePresignTvm.js:32]: (24 * 60 * 60) is way clearer, also/or a comment saying `// limit expiry between 2 seconds and 24 hours`

File: lib/impl/AzurePresignTvm.js [hunk 3]
```diff
@@ -29,7 +29,10 @@ class AzurePresignTvm extends Tvm {
...
+    const limitLength = new RegExp('^.{1,3}$')
+    const limitChars = new RegExp('^[rwd]*$')
```
Comment (purplecabbage) [inline lib/impl/AzurePresignTvm.js:34]: imho, it should be law that regex has a comment, even if it is fairly simple
`// length 1-3`
`// only 'r', 'w' or 'd'`

File: lib/impl/AzurePresignTvm.js [hunk 4]
```diff
@@ -42,10 +45,10 @@ class AzurePresignTvm extends Tvm {
     const containerName = Tvm._hash(params.owNamespace)
 
     // generate SAS token
-    const expiryTime = new Date()
-    expiryTime.setSeconds(expiryTime.getSeconds() + params.expiryInSeconds)
+    const expiryTime = new Date(new Date().getTime() + (1000 * params.expiryInSeconds))
```
Comment (purplecabbage) [inline lib/impl/AzurePresignTvm.js:48]: `Date.now()` is the new `new Date().getTime()`

**Review by purplecabbage — APPROVED** [review]
added a few minor nitpicks, but this looks good!
:shipit:

**Pull request merged**